components/viaduct/src/backend/ffi.rs

```diff
@@ -38,78 +38,77 @@ macro_rules! backend_error {
     }};
 }
 
-pub fn send(request: crate::Request) -> Result<crate::Response, Error> {
-    use ffi_support::IntoFfi;
-    use prost::Message;
-    super::note_backend("FFI (trusted)");
-
-    let method = request.method;
-    let fetch = callback_holder::get_callback().ok_or_else(|| Error::BackendNotInitialized)?;
-    let proto_req: msg_types::Request = request.into();
-    let buf = proto_req.into_ffi_value();
-    let response = unsafe { fetch(buf) };
-    // This way we'll Drop it if we panic, unlike if we just got a slice into
-    // it. Besides, we already own it.
-    let response_bytes = response.into_vec();
-
-    let response: msg_types::Response = match Message::decode(response_bytes.as_slice()) {
-        Ok(v) => v,
-        Err(e) => {
-            panic!(
-                "Failed to parse protobuf returned from fetch callback! {}",
-                e
+pub struct FFIBackend;
```

Comment:
Nit: Rust style is generally not to capitalize abbreviations, so `FfiBackend` (Avoids XMLHttpRequest-type issues).